apps/aehttp/test/aehttp_contracts_SUITE.erl

```diff
@@ -482,60 +464,185 @@ counter_contract(Config) ->
+    #{<<"value">> := All} = decode_data(<<"list(string)">>, Ret3),
+    ct:pal("All ~p\n", [All]),
+
+    %% Pop the values.
+    <<"four">> = call_pop(NodeName, APubkey, APrivkey, EncodedContractPubkey),
+    <<"three">> = call_pop(NodeName, BPubkey, BPrivkey, EncodedContractPubkey),
+    <<"two">> = call_pop(NodeName, CPubkey, CPrivkey, EncodedContractPubkey),
+    <<"one">> = call_pop(NodeName, DPubkey, DPrivkey, EncodedContractPubkey),
+
+    %% The resulting stack is empty.
+    {Ret4,_} = call_compute_func(NodeName, APubkey, APrivkey,
+                                 EncodedContractPubkey, <<"size">>, <<"()">>),
+    #{<<"value">> := 0} = decode_data(<<"int">>, Ret4),
+
+    ok.
+
+call_pop(NodeName, Pubkey, Privkey, EncodedContractPubkey) ->
+    {Value,_Return} =
+        call_compute_func(NodeName, Pubkey, Privkey, EncodedContractPubkey,
+                          <<"pop">>, <<"()">>),
+    #{<<"value">> := String} = decode_data(<<"string">>, Value),
+    String.
+
+%% polymorphism_test_contract(Config)
+%%  Check the polymorphism_test contract.
+%%  This does not work yet.
+
+polymorphism_test_contract(Config) ->
+    NodeName = proplists:get_value(node_name, Config),
+    %% Get account information.
+    #{acc_a := #{pub_key := APubkey,
+                 priv_key := APrivkey}} = proplists:get_value(accounts, Config),
+
+    %% Make sure accounts have enough tokens.
+    _ABal0 = ensure_balance(APubkey, 500000),
+    {ok,[_]} = aecore_suite_utils:mine_key_blocks(NodeName, 1),
+
+    %% Compile test contract "polymorphism_test.aes".
+    Code = compile_test_contract("polymorphism_test"),
+
+    %% Initialise contract owned by Alice.
+    {EncodedContractPubkey,_,_} =
+       create_compute_contract(NodeName, APubkey, APrivkey, Code, <<"()">>),
+
+    %% Test the polymorphism.
+    call_compute_func(NodeName, APubkey, APrivkey, EncodedContractPubkey,
+                      <<"foo">>, <<"()">>),
+    call_compute_func(NodeName, APubkey, APrivkey, EncodedContractPubkey,
+                      <<"bar">>, <<"()">>),
+
+    ok.
+
+%% factorial_contract(Config)
+%%  Check the factorial contract.
+
+factorial_contract(Config) ->
+    NodeName = proplists:get_value(node_name, Config),
+    %% Get account information.
+    #{acc_a := #{pub_key := APubkey,
+                 priv_key := APrivkey}} = proplists:get_value(accounts, Config),
+
+    %% Make sure accounts have enough tokens.
+    _ABal0 = ensure_balance(APubkey, 500000),
+    {ok,[_]} = aecore_suite_utils:mine_key_blocks(NodeName, 1),
+
+    %% Compile test contract "factorial.aes".
+    Code = compile_test_contract("factorial"),
+
+    %% Initialise contract owned by Alice.
+    {EncodedContractPubkey,DecodedContractPubkey,_} =
+       create_compute_contract(NodeName, APubkey, APrivkey, Code, <<"(0)">>),
+
+    %% Set worker contract.
+    call_compute_func(NodeName, APubkey, APrivkey, EncodedContractPubkey,
+                      <<"set_worker">>,
```

Comment:
I see this is needed for overwriting the worker 0 set at contract creation. Ok.

Reply:
Yes, I found no better way of getting the contract pubkey into the contract itself, as it does not exist before the contract is created. Maybe a comment would be in place here.